src/chain.js

```diff
@@ -78,10 +83,22 @@ const chain = ({ request, source, head, handlerOpts, server, isPlain }) => {
 
         source.pipe(socket);
         socket.pipe(source);
+
+        // Once target socket closes forcibly, the source socket gets paused.
+        // We need to enable flowing, otherwise the socket would remain open indefinitely.
+        // Nothing would consume the data, we just want to close the socket.
+        source.on('close', () => {
+            socket.resume();
```

Comment:
The names `source` and `socket` look visually quite the same, so it's easy to miss the difference.

Reply:
I'll rename this to `sourceSocket` and `targetSocket`.